src/nvim/eval.c

```diff
@@ -12064,26 +12064,51 @@ static void get_maparg(typval_T *argvars, typval_T *rettv, int exact)
     tv_dict_alloc_ret(rettv);
     if (rhs != NULL) {
       // Return a dictionary.
-      char_u *lhs = str2special_save(mp->m_keys, true);
-      char *const mapmode = map_mode_to_chars(mp->m_mode);
-      dict_T *dict = rettv->vval.v_dict;
-
-      tv_dict_add_str(dict, S_LEN("lhs"), (const char *)lhs);
-      tv_dict_add_str(dict, S_LEN("rhs"), (const char *)mp->m_orig_str);
-      tv_dict_add_nr(dict, S_LEN("noremap"), mp->m_noremap ? 1 : 0);
-      tv_dict_add_nr(dict, S_LEN("expr"),  mp->m_expr ? 1 : 0);
-      tv_dict_add_nr(dict, S_LEN("silent"), mp->m_silent ? 1 : 0);
-      tv_dict_add_nr(dict, S_LEN("sid"), (varnumber_T)mp->m_script_ID);
-      tv_dict_add_nr(dict, S_LEN("buffer"), (varnumber_T)buffer_local);
-      tv_dict_add_nr(dict, S_LEN("nowait"), mp->m_nowait ? 1 : 0);
-      tv_dict_add_str(dict, S_LEN("mode"), mapmode);
-
-      xfree(lhs);
-      xfree(mapmode);
+      mapblock_fill_dict(rettv->vval.v_dict, mp, buffer_local, true);
     }
   }
 }
 
+/// Fill a dictionary with all applicable maparg() like dictionaries
```

Comment:
It is strange: GH does not highlight this file starting from this line. All black.